lib/scrub/scrub_runner.rb

```diff
@@ -0,0 +1,125 @@
+# A fully automated scrubber would:
+# Check org directories for new org-submitted files
+# Compare remote files to processed files
+# Copy new files to work directory
+# Validate and scrub files into loadable format
+# Load files into db
+# Upload report to org dir
+# ... should be able to do this for any number of orgs/files, at a whim or by cron,
+# ideally using a queue or pool of workers.
+
+require "data_sources/directory_locator"
+require "data_sources/ht_organizations"
+require "fileutils"
+require "loader/file_loader"
+require "loader/holding_loader"
+require "scrub/autoscrub"
+require "scrub/chunker"
+require "utils/file_transfer"
+
+# Example:
+# runner = Scrub::ScrubRunner.new(ORG)
+# runner.run
+
+module Scrub
+  # Scrubs and loads any new org-uploaded files.
+  class ScrubRunner
+    def initialize(organization)
+      @organization = organization
+      @ft = Utils::FileTransfer.new
+      validate
+    end
+
+    def validate
+      if Settings.local_member_data.nil?
+        raise "Need Settings.local_member_data to be set"
+      end
+      if Settings.remote_member_data.nil?
+        raise "Need Settings.remote_member_data to be set"
+      end
+      if @organization.nil?
+        raise "Need @organization to be set"
+      end
+    end
+
+    def run
+      puts "Running org #{@organization}."
+      new_files = check_new_files
+      puts "Found #{new_files.size} new files: #{new_files.join(", ")}."
+      new_files.each do |new_file|
+        run_file(new_file)
+      end
+    end
+
+    def run_file(file)
+      puts "Running file #{file}"
+      downloaded_file = download_to_work_dir(file)
+      scrubber = Scrub::AutoScrub.new(downloaded_file)
+      scrubber.run
+      scrubber.out_files.each do |scrubber_out_file|
+        puts "Ready to split #{scrubber_out_file} into chunks"
```

Comment:
Consider extracting the contents of the block into a method on Chunker, perhaps like:

```ruby
scrubber.out_files.each do |scrubber_out_file|
  Scrub::Chunker.new(glob: file, chunk_count: 4, out_ext: "ndj").chunk_and_load
  upload_to_member(scrubber.logger_path)
end

class Scrub::Chunker
  def chunk_and_load(file)
   Services.logger.info("Ready to split #{file} into chunks")
   run
   load_chunks
   cleanup!
  end

  def load_chunks
    chunks.each do |chunk|
     Services.logger.info("Load chunk #{chunk}")
     Loader::FileLoader.new(batch_loader: Loader::HoldingLoader.for(chunk)).load(chunk)
    end
  end
end
```

Reply:
I see what you mean, but I like it better having the coordination done by (the already necessarily more complex) `ScrubRunner`, and leaving `Chunker` with a more limited scope.

Reply:
:+1: